internal/exec/util/selinux.go

```diff
@@ -84,3 +84,13 @@ func (ut Util) RelabelFiles(patterns []string) error {
 	}
 	return nil
 }
+
+func FileSystemIsReadOnly(path string) bool {
+	//open file for writing
+	file, err := os.OpenFile(path, os.O_WRONLY, 0666)
+	if err != nil {
+		return os.IsPermission(err)
```

Comment:
Wasn't exactly sure how to test when writing this at first, 

but after reading this i think this is what your looking for:
`func isReadOnlyFileSystem(err error) bool {
	if pe, ok := err.(*os.PathError); ok {
		if errno, ok := pe.Err.(syscall.Errno); ok {
			return errno == syscall.EROFS || errno == syscall.EISDIR
		}
	}
	return false
}`

This function will return true if the directory is attempted to be open as a file or if the fs is read only

Reply:
That's true, it will.  But you're not getting EISDIR as a valid indication that the filesystem is read-only, you're getting it because your code is wrong.  It doesn't make sense to try to open a directory as though it were a file.

You could try to create a temporary file _inside_ the directory.  But that still has a bit of a code smell, because you'd be changing something solely to see whether you have permission to change it.  The better approach is to query the filesystem with [`statfs(2)`](https://man7.org/linux/man-pages/man2/statfs.2.html) and check for the `ST_READONLY` flag.  Unfortunately, while the `statfs` syscall is available in `x/sys`, the value of `ST_READONLY` doesn't seem to be available anywhere in the Go standard libraries, so we'd need a bit of C glue for this.

But luckily none of that is relevant here, since we're [not going to be doing this check](https://github.com/coreos/ignition/pull/1608#discussion_r1222362864).